server/ServerFSM.cpp

```diff
@@ -1273,11 +1273,40 @@ sc::result MPLobby::react(const LobbyUpdate& msg) {
             psd.second.m_save_game_empire_id = ALL_EMPIRES;
         }
 
+        // remove all AIs
+        m_lobby_data->m_players.remove_if([](const std::pair<int, PlayerSetupData>& plr) {
+            return plr.second.m_client_type == Networking::CLIENT_TYPE_AI_PLAYER;
+        });
+        m_ai_next_index = 1;
+
         // refresh save game empire data
         boost::filesystem::path save_dir(GetServerSaveDir());
+        std::vector<PlayerSaveHeaderData> player_save_header_data;
         try {
             LoadEmpireSaveGameData((save_dir / m_lobby_data->m_save_game).string(),
-                                   m_lobby_data->m_save_game_empire_data);
+                                   m_lobby_data->m_save_game_empire_data,
+                                   player_save_header_data);
+
+            for (const auto& pshd : player_save_header_data) {
+                if (pshd.m_client_type == Networking::CLIENT_TYPE_AI_PLAYER) {
+                    const auto& empire_data_it = m_lobby_data->m_save_game_empire_data.find(pshd.m_empire_id);
+                    if (empire_data_it != m_lobby_data->m_save_game_empire_data.end()) {
+                        PlayerSetupData player_setup_data;
+                        player_setup_data.m_player_id =     Networking::INVALID_PLAYER_ID;
+                        // don't use original names to prevent collision with manually added AIs
+                        player_setup_data.m_player_name =   UserString("AI_PLAYER") + "_" + std::to_string(m_ai_next_index++);
```

Comment:
would be better to use a substitution rather than appending for combining "AI Player" with a number, but that can be done separately.

Reply:
It's a same expression used in other functions where new AI was appended.

Reply:
I realize... doesn't mean it can't be improved for better multilingual compatibility.

Reply:
I suppose it better to do in other PR.

Reply:
yes; that's what I meant by "can be done separately"